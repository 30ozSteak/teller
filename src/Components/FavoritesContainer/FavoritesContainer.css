.container-box {
  overflow: hidden;
  height: 11rem;
}
.container-box:before {
  width: 23rem;
  height: 1%;
  content: "";
  border-radius: 0 0 5px 5px;
  position: absolute;
  /* top: 11rem; */
  background-color: #aa78e6;
  left: 0;
  right: 0;
  margin: 0 auto;
  z-index: 20;
  top: 11rem;
}

.container-box:after {
  width: 85%;
  height: 1.5%;
  content: "";
  border-radius: 0 0 5px 5px;
  position: absolute;
  top: 11.15rem;
  background-color: #c5a3ee;
  left: 0;
  right: 0;
  margin: 0 auto;
  z-index: 15;
}
