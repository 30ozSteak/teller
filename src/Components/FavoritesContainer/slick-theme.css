/* .container {
  margin: 0 auto;
  width: 80%;
} */

/* ignore this for now */
