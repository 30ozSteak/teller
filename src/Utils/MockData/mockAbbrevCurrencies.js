export const mockAbbrevCurrencies = [
  {
    "symbol": "BTC",
    "price": 3586.16,
    "percent_change": 0.83,
    "name": "Bitcoin",
    "rank": "1",
  },
  {
    "symbol": "XRP",
    "price": 0.33,
    "percent_change": 3.94,
    "name": "XRP",
    "rank": "2",
  },
  {
    "symbol": "ETH",
    "price": 95.07,
    "percent_change": 0.20,
    "name": "Ethereum",
    "rank": "3",
  },
  {
    "symbol": "EOS",
    "price": 2.54,
    "percent_change": 6.87,
    "name": "EOS",
    "rank": "4",
  },
  {
    "symbol": "XLM",
    "price": 0.11,
    "percent_change": -0.58,
    "name": "Stellar",
    "rank": "5",
  },
  {
    "symbol": "USDT",
    "price": 1.02,
    "percent_change": -0.37,
    "name": "Tether",
    "rank": "6",
  },
  {
    "symbol": "BCH",
    "price": 100.65,
    "percent_change": 12.32,
    "name": "Bitcoin Cash",
    "rank": "7",
  },
  {
    "symbol": "LTC",
    "price": 28.99,
    "percent_change": -1.97,
    "name": "Litecoin",
    "rank": "8",
  },
  {
    "symbol": "BSV",
    "price": 81.84,
    "percent_change": -1.04,
    "name": "Bitcoin SV",
    "rank": "9",
  },
  {
    "symbol": "TRX",
    "price": 0.01,
    "percent_change": 0.17,
    "name": "TRON",
    "rank": "10",
  }
]