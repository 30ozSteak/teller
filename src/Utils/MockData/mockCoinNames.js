export const mockCoinNames = [
  "Bitcoin",
  "XRP",
  "Ethereum",
  "EOS",
  "Stellar",
  "Tether",
  "Bitcoin Cash",
  "Litecoin",
  "Bitcoin SV",
  "TRON"
]