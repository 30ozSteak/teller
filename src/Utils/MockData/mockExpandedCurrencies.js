export const mockExpandedCurrencies = [
    {
        "symbol": "BTC",
        "price": 3586.16,
        "percent_change": 0.83,
        "name": "Bitcoin",
        "rank": "1",
        "supply": 17429837.00,
        "market_cap": 62506159630.72,
    },
    {
        "symbol": "XRP",
        "price": 0.33,
        "percent_change": 3.94,
        "name": "XRP",
        "rank": "2",
        "supply": 40762365544.00,
        "market_cap": 13612140064.75,
    },
    {
        "symbol": "ETH",
        "price": 95.07,
        "percent_change": 0.20,
        "name": "Ethereum",
        "rank": "3",
        "supply": 103875264.97,
        "market_cap": 9875418262.80,
    },
    {
        "symbol": "EOS",
        "price": 2.54,
        "percent_change": 6.87,
        "name": "EOS",
        "rank": "4",
        "supply": 906245117.60,
        "market_cap": 2299890068.43,
    },
    {
        "symbol": "XLM",
        "price": 0.11,
        "percent_change": -0.58,
        "name": "Stellar",
        "rank": "5",
        "supply": 19171055431.43,
        "market_cap": 2101330537.32,
    },
    {
        "symbol": "USDT",
        "price": 1.02,
        "percent_change": -0.37,
        "name": "Tether",
        "rank": "6",
        "supply": 1856421735.62,
        "market_cap": 1884620732.28,
    },
    {
        "symbol": "BCH",
        "price": 100.65,
        "percent_change": 12.32,
        "name": "Bitcoin Cash",
        "rank": "7",
        "supply": 17517750.00,
        "market_cap": 1763235900.72,
    },
    {
        "symbol": "LTC",
        "price": 28.99,
        "percent_change": -1.97,
        "name": "Litecoin",
        "rank": "8",
        "supply": 59624662.94,
        "market_cap": 1728272479.52,
    },
    {
        "symbol": "BSV",
        "price": 81.84,
        "percent_change": -1.04,
        "name": "Bitcoin SV",
        "rank": "9",
        "supply": 17517010.94,
        "market_cap": 1433656314.37,
    },
    {
        "symbol": "TRX",
        "price": 0.01,
        "percent_change": 0.17,
        "name": "TRON",
        "rank": "10",
        "supply": 66464300491.53,
        "market_cap": 951584289.18,
    }
]