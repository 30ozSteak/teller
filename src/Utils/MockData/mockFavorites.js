export const mockFavorites = [
  {name: "DopeCoin", price: 1000000, percent_change: 1.00},
  {name: "DoperCoin", price: 1000000, percent_change: 1.00},
  {name: "DopestCoin", price: 1000000, percent_change: 1.00},
  {name: "Even DoperCoin", price: 1000000, percent_change: 1.00},
  {name: "DopestCoinOfAll", price: 1000000, percent_change: 1.00},
]