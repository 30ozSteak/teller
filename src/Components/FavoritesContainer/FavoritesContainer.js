import React, { Component } from "react";

import Slider from "react-slick";
import FavoriteCurrencies from "../FavoriteCurrencies/FavoriteCurrencies";
import settings from "../Settings/Settings";
import "./slick-theme.css";
import "./FavoritesContainer.css";
import TellerAI from "../tellerAI/tellerAI";

class FavoritesContainer extends Component {
  constructor({
      props,
      favorites,
      addToFavorites,
      removeFromFavorites,
      abbrevCurrencies,
      setFilter
    }) {
    super(props);
    this.state = {
      expanded: false
    };
  }

  render() {
    const { 
      favorites, 
      removeFromFavorites, 
      addToFavorites } = this.props;
    const favoriteCurrencies = favorites.map(favorite => {
      return (
        <FavoriteCurrencies
          key={favorite.id}
          favorite={favorite}
          removeFromFavorites={removeFromFavorites}
          addToFavorites={addToFavorites}
        />
      );
    });

    return (
      <div className="container-box">
        <div className="container">
          <Slider {...settings}>{favoriteCurrencies}</Slider>
        </div>
        {/* <TellerAI /> */}
      </div>
    );
  }
}

export default FavoritesContainer;
