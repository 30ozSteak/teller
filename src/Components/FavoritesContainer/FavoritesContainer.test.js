import React from 'react'
import { shallow } from 'enzyme'
import FavoritesContainer from './FavoritesContainer'
import { mockCurrencies } from '../../Utils/MockData/mockCurrencies'

describe('FavoritesContainer Component', () => {
  let wrapper
  let favorites = mockCurrencies.slice(0, 5)

  beforeEach(() => {
    wrapper = shallow(<FavoritesContainer favorites={favorites}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})