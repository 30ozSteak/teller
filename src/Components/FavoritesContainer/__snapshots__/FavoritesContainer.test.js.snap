// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesContainer Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoritesContainer
    favorites={
      Array [
        Object {
          "market_cap_usd": "62506159630.7222615731538755",
          "max_supply": "21000000.0000000000000000",
          "name": "Bitcoin",
          "percent_change_24_hr": "0.8320854828625978",
          "price_usd": "3586.1585871814097615",
          "rank": "1",
          "supply": "17429837.0000000000000000",
          "symbol": "BTC",
        },
        Object {
          "market_cap_usd": "13612140064.7471135264064432",
          "max_supply": "100000000000.0000000000000000",
          "name": "XRP",
          "percent_change_24_hr": "3.9433058589033265",
          "price_usd": "0.3339389135808078",
          "rank": "2",
          "supply": "40762365544.0000000000000000",
          "symbol": "XRP",
        },
        Object {
          "market_cap_usd": "9875418262.8035156239002594",
          "max_supply": null,
          "name": "Ethereum",
          "percent_change_24_hr": "0.2048869680495625",
          "price_usd": "95.0699694086438060",
          "rank": "3",
          "supply": "103875264.9678000000000000",
          "symbol": "ETH",
        },
        Object {
          "market_cap_usd": "2299890068.4280425642158096",
          "max_supply": null,
          "name": "EOS",
          "percent_change_24_hr": "6.8707627494412566",
          "price_usd": "2.5378234031415460",
          "rank": "4",
          "supply": "906245117.6000000000000000",
          "symbol": "EOS",
        },
        Object {
          "market_cap_usd": "2101330537.3210911914502786",
          "max_supply": null,
          "name": "Stellar",
          "percent_change_24_hr": "-0.5797575094501702",
          "price_usd": "0.1096095384438811",
          "rank": "5",
          "supply": "19171055431.4299000000000000",
          "symbol": "XLM",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <Slider
          arrows={false}
          buttons={false}
          dots={true}
          infinite={true}
          initialSlide={1}
          slidesToScroll={1}
          slidesToShow={1}
          swipe={true}
          vertical={true}
          verticalSwiping={true}
        >
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "62506159630.7222615731538755",
                "max_supply": "21000000.0000000000000000",
                "name": "Bitcoin",
                "percent_change_24_hr": "0.8320854828625978",
                "price_usd": "3586.1585871814097615",
                "rank": "1",
                "supply": "17429837.0000000000000000",
                "symbol": "BTC",
              }
            }
          />
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "13612140064.7471135264064432",
                "max_supply": "100000000000.0000000000000000",
                "name": "XRP",
                "percent_change_24_hr": "3.9433058589033265",
                "price_usd": "0.3339389135808078",
                "rank": "2",
                "supply": "40762365544.0000000000000000",
                "symbol": "XRP",
              }
            }
          />
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "9875418262.8035156239002594",
                "max_supply": null,
                "name": "Ethereum",
                "percent_change_24_hr": "0.2048869680495625",
                "price_usd": "95.0699694086438060",
                "rank": "3",
                "supply": "103875264.9678000000000000",
                "symbol": "ETH",
              }
            }
          />
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "2299890068.4280425642158096",
                "max_supply": null,
                "name": "EOS",
                "percent_change_24_hr": "6.8707627494412566",
                "price_usd": "2.5378234031415460",
                "rank": "4",
                "supply": "906245117.6000000000000000",
                "symbol": "EOS",
              }
            }
          />
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "2101330537.3210911914502786",
                "max_supply": null,
                "name": "Stellar",
                "percent_change_24_hr": "-0.5797575094501702",
                "price_usd": "0.1096095384438811",
                "rank": "5",
                "supply": "19171055431.4299000000000000",
                "symbol": "XLM",
              }
            }
          />
        </Slider>
      </div>,
      "className": "container-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Slider
          arrows={false}
          buttons={false}
          dots={true}
          infinite={true}
          initialSlide={1}
          slidesToScroll={1}
          slidesToShow={1}
          swipe={true}
          vertical={true}
          verticalSwiping={true}
        >
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "62506159630.7222615731538755",
                "max_supply": "21000000.0000000000000000",
                "name": "Bitcoin",
                "percent_change_24_hr": "0.8320854828625978",
                "price_usd": "3586.1585871814097615",
                "rank": "1",
                "supply": "17429837.0000000000000000",
                "symbol": "BTC",
              }
            }
          />
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "13612140064.7471135264064432",
                "max_supply": "100000000000.0000000000000000",
                "name": "XRP",
                "percent_change_24_hr": "3.9433058589033265",
                "price_usd": "0.3339389135808078",
                "rank": "2",
                "supply": "40762365544.0000000000000000",
                "symbol": "XRP",
              }
            }
          />
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "9875418262.8035156239002594",
                "max_supply": null,
                "name": "Ethereum",
                "percent_change_24_hr": "0.2048869680495625",
                "price_usd": "95.0699694086438060",
                "rank": "3",
                "supply": "103875264.9678000000000000",
                "symbol": "ETH",
              }
            }
          />
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "2299890068.4280425642158096",
                "max_supply": null,
                "name": "EOS",
                "percent_change_24_hr": "6.8707627494412566",
                "price_usd": "2.5378234031415460",
                "rank": "4",
                "supply": "906245117.6000000000000000",
                "symbol": "EOS",
              }
            }
          />
          <FavoriteCurrencies
            favorite={
              Object {
                "market_cap_usd": "2101330537.3210911914502786",
                "max_supply": null,
                "name": "Stellar",
                "percent_change_24_hr": "-0.5797575094501702",
                "price_usd": "0.1096095384438811",
                "rank": "5",
                "supply": "19171055431.4299000000000000",
                "symbol": "XLM",
              }
            }
          />
        </Slider>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrows": false,
          "buttons": false,
          "children": Array [
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "62506159630.7222615731538755",
                  "max_supply": "21000000.0000000000000000",
                  "name": "Bitcoin",
                  "percent_change_24_hr": "0.8320854828625978",
                  "price_usd": "3586.1585871814097615",
                  "rank": "1",
                  "supply": "17429837.0000000000000000",
                  "symbol": "BTC",
                }
              }
            />,
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "13612140064.7471135264064432",
                  "max_supply": "100000000000.0000000000000000",
                  "name": "XRP",
                  "percent_change_24_hr": "3.9433058589033265",
                  "price_usd": "0.3339389135808078",
                  "rank": "2",
                  "supply": "40762365544.0000000000000000",
                  "symbol": "XRP",
                }
              }
            />,
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "9875418262.8035156239002594",
                  "max_supply": null,
                  "name": "Ethereum",
                  "percent_change_24_hr": "0.2048869680495625",
                  "price_usd": "95.0699694086438060",
                  "rank": "3",
                  "supply": "103875264.9678000000000000",
                  "symbol": "ETH",
                }
              }
            />,
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "2299890068.4280425642158096",
                  "max_supply": null,
                  "name": "EOS",
                  "percent_change_24_hr": "6.8707627494412566",
                  "price_usd": "2.5378234031415460",
                  "rank": "4",
                  "supply": "906245117.6000000000000000",
                  "symbol": "EOS",
                }
              }
            />,
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "2101330537.3210911914502786",
                  "max_supply": null,
                  "name": "Stellar",
                  "percent_change_24_hr": "-0.5797575094501702",
                  "price_usd": "0.1096095384438811",
                  "rank": "5",
                  "supply": "19171055431.4299000000000000",
                  "symbol": "XLM",
                }
              }
            />,
          ],
          "dots": true,
          "infinite": true,
          "initialSlide": 1,
          "slidesToScroll": 1,
          "slidesToShow": 1,
          "swipe": true,
          "vertical": true,
          "verticalSwiping": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addToFavorites": undefined,
              "favorite": Object {
                "market_cap_usd": "62506159630.7222615731538755",
                "max_supply": "21000000.0000000000000000",
                "name": "Bitcoin",
                "percent_change_24_hr": "0.8320854828625978",
                "price_usd": "3586.1585871814097615",
                "rank": "1",
                "supply": "17429837.0000000000000000",
                "symbol": "BTC",
              },
              "removeFromFavorites": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addToFavorites": undefined,
              "favorite": Object {
                "market_cap_usd": "13612140064.7471135264064432",
                "max_supply": "100000000000.0000000000000000",
                "name": "XRP",
                "percent_change_24_hr": "3.9433058589033265",
                "price_usd": "0.3339389135808078",
                "rank": "2",
                "supply": "40762365544.0000000000000000",
                "symbol": "XRP",
              },
              "removeFromFavorites": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addToFavorites": undefined,
              "favorite": Object {
                "market_cap_usd": "9875418262.8035156239002594",
                "max_supply": null,
                "name": "Ethereum",
                "percent_change_24_hr": "0.2048869680495625",
                "price_usd": "95.0699694086438060",
                "rank": "3",
                "supply": "103875264.9678000000000000",
                "symbol": "ETH",
              },
              "removeFromFavorites": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addToFavorites": undefined,
              "favorite": Object {
                "market_cap_usd": "2299890068.4280425642158096",
                "max_supply": null,
                "name": "EOS",
                "percent_change_24_hr": "6.8707627494412566",
                "price_usd": "2.5378234031415460",
                "rank": "4",
                "supply": "906245117.6000000000000000",
                "symbol": "EOS",
              },
              "removeFromFavorites": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addToFavorites": undefined,
              "favorite": Object {
                "market_cap_usd": "2101330537.3210911914502786",
                "max_supply": null,
                "name": "Stellar",
                "percent_change_24_hr": "-0.5797575094501702",
                "price_usd": "0.1096095384438811",
                "rank": "5",
                "supply": "19171055431.4299000000000000",
                "symbol": "XLM",
              },
              "removeFromFavorites": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <Slider
            arrows={false}
            buttons={false}
            dots={true}
            infinite={true}
            initialSlide={1}
            slidesToScroll={1}
            slidesToShow={1}
            swipe={true}
            vertical={true}
            verticalSwiping={true}
          >
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "62506159630.7222615731538755",
                  "max_supply": "21000000.0000000000000000",
                  "name": "Bitcoin",
                  "percent_change_24_hr": "0.8320854828625978",
                  "price_usd": "3586.1585871814097615",
                  "rank": "1",
                  "supply": "17429837.0000000000000000",
                  "symbol": "BTC",
                }
              }
            />
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "13612140064.7471135264064432",
                  "max_supply": "100000000000.0000000000000000",
                  "name": "XRP",
                  "percent_change_24_hr": "3.9433058589033265",
                  "price_usd": "0.3339389135808078",
                  "rank": "2",
                  "supply": "40762365544.0000000000000000",
                  "symbol": "XRP",
                }
              }
            />
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "9875418262.8035156239002594",
                  "max_supply": null,
                  "name": "Ethereum",
                  "percent_change_24_hr": "0.2048869680495625",
                  "price_usd": "95.0699694086438060",
                  "rank": "3",
                  "supply": "103875264.9678000000000000",
                  "symbol": "ETH",
                }
              }
            />
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "2299890068.4280425642158096",
                  "max_supply": null,
                  "name": "EOS",
                  "percent_change_24_hr": "6.8707627494412566",
                  "price_usd": "2.5378234031415460",
                  "rank": "4",
                  "supply": "906245117.6000000000000000",
                  "symbol": "EOS",
                }
              }
            />
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "2101330537.3210911914502786",
                  "max_supply": null,
                  "name": "Stellar",
                  "percent_change_24_hr": "-0.5797575094501702",
                  "price_usd": "0.1096095384438811",
                  "rank": "5",
                  "supply": "19171055431.4299000000000000",
                  "symbol": "XLM",
                }
              }
            />
          </Slider>
        </div>,
        "className": "container-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Slider
            arrows={false}
            buttons={false}
            dots={true}
            infinite={true}
            initialSlide={1}
            slidesToScroll={1}
            slidesToShow={1}
            swipe={true}
            vertical={true}
            verticalSwiping={true}
          >
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "62506159630.7222615731538755",
                  "max_supply": "21000000.0000000000000000",
                  "name": "Bitcoin",
                  "percent_change_24_hr": "0.8320854828625978",
                  "price_usd": "3586.1585871814097615",
                  "rank": "1",
                  "supply": "17429837.0000000000000000",
                  "symbol": "BTC",
                }
              }
            />
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "13612140064.7471135264064432",
                  "max_supply": "100000000000.0000000000000000",
                  "name": "XRP",
                  "percent_change_24_hr": "3.9433058589033265",
                  "price_usd": "0.3339389135808078",
                  "rank": "2",
                  "supply": "40762365544.0000000000000000",
                  "symbol": "XRP",
                }
              }
            />
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "9875418262.8035156239002594",
                  "max_supply": null,
                  "name": "Ethereum",
                  "percent_change_24_hr": "0.2048869680495625",
                  "price_usd": "95.0699694086438060",
                  "rank": "3",
                  "supply": "103875264.9678000000000000",
                  "symbol": "ETH",
                }
              }
            />
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "2299890068.4280425642158096",
                  "max_supply": null,
                  "name": "EOS",
                  "percent_change_24_hr": "6.8707627494412566",
                  "price_usd": "2.5378234031415460",
                  "rank": "4",
                  "supply": "906245117.6000000000000000",
                  "symbol": "EOS",
                }
              }
            />
            <FavoriteCurrencies
              favorite={
                Object {
                  "market_cap_usd": "2101330537.3210911914502786",
                  "max_supply": null,
                  "name": "Stellar",
                  "percent_change_24_hr": "-0.5797575094501702",
                  "price_usd": "0.1096095384438811",
                  "rank": "5",
                  "supply": "19171055431.4299000000000000",
                  "symbol": "XLM",
                }
              }
            />
          </Slider>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrows": false,
            "buttons": false,
            "children": Array [
              <FavoriteCurrencies
                favorite={
                  Object {
                    "market_cap_usd": "62506159630.7222615731538755",
                    "max_supply": "21000000.0000000000000000",
                    "name": "Bitcoin",
                    "percent_change_24_hr": "0.8320854828625978",
                    "price_usd": "3586.1585871814097615",
                    "rank": "1",
                    "supply": "17429837.0000000000000000",
                    "symbol": "BTC",
                  }
                }
              />,
              <FavoriteCurrencies
                favorite={
                  Object {
                    "market_cap_usd": "13612140064.7471135264064432",
                    "max_supply": "100000000000.0000000000000000",
                    "name": "XRP",
                    "percent_change_24_hr": "3.9433058589033265",
                    "price_usd": "0.3339389135808078",
                    "rank": "2",
                    "supply": "40762365544.0000000000000000",
                    "symbol": "XRP",
                  }
                }
              />,
              <FavoriteCurrencies
                favorite={
                  Object {
                    "market_cap_usd": "9875418262.8035156239002594",
                    "max_supply": null,
                    "name": "Ethereum",
                    "percent_change_24_hr": "0.2048869680495625",
                    "price_usd": "95.0699694086438060",
                    "rank": "3",
                    "supply": "103875264.9678000000000000",
                    "symbol": "ETH",
                  }
                }
              />,
              <FavoriteCurrencies
                favorite={
                  Object {
                    "market_cap_usd": "2299890068.4280425642158096",
                    "max_supply": null,
                    "name": "EOS",
                    "percent_change_24_hr": "6.8707627494412566",
                    "price_usd": "2.5378234031415460",
                    "rank": "4",
                    "supply": "906245117.6000000000000000",
                    "symbol": "EOS",
                  }
                }
              />,
              <FavoriteCurrencies
                favorite={
                  Object {
                    "market_cap_usd": "2101330537.3210911914502786",
                    "max_supply": null,
                    "name": "Stellar",
                    "percent_change_24_hr": "-0.5797575094501702",
                    "price_usd": "0.1096095384438811",
                    "rank": "5",
                    "supply": "19171055431.4299000000000000",
                    "symbol": "XLM",
                  }
                }
              />,
            ],
            "dots": true,
            "infinite": true,
            "initialSlide": 1,
            "slidesToScroll": 1,
            "slidesToShow": 1,
            "swipe": true,
            "vertical": true,
            "verticalSwiping": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addToFavorites": undefined,
                "favorite": Object {
                  "market_cap_usd": "62506159630.7222615731538755",
                  "max_supply": "21000000.0000000000000000",
                  "name": "Bitcoin",
                  "percent_change_24_hr": "0.8320854828625978",
                  "price_usd": "3586.1585871814097615",
                  "rank": "1",
                  "supply": "17429837.0000000000000000",
                  "symbol": "BTC",
                },
                "removeFromFavorites": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addToFavorites": undefined,
                "favorite": Object {
                  "market_cap_usd": "13612140064.7471135264064432",
                  "max_supply": "100000000000.0000000000000000",
                  "name": "XRP",
                  "percent_change_24_hr": "3.9433058589033265",
                  "price_usd": "0.3339389135808078",
                  "rank": "2",
                  "supply": "40762365544.0000000000000000",
                  "symbol": "XRP",
                },
                "removeFromFavorites": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addToFavorites": undefined,
                "favorite": Object {
                  "market_cap_usd": "9875418262.8035156239002594",
                  "max_supply": null,
                  "name": "Ethereum",
                  "percent_change_24_hr": "0.2048869680495625",
                  "price_usd": "95.0699694086438060",
                  "rank": "3",
                  "supply": "103875264.9678000000000000",
                  "symbol": "ETH",
                },
                "removeFromFavorites": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addToFavorites": undefined,
                "favorite": Object {
                  "market_cap_usd": "2299890068.4280425642158096",
                  "max_supply": null,
                  "name": "EOS",
                  "percent_change_24_hr": "6.8707627494412566",
                  "price_usd": "2.5378234031415460",
                  "rank": "4",
                  "supply": "906245117.6000000000000000",
                  "symbol": "EOS",
                },
                "removeFromFavorites": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addToFavorites": undefined,
                "favorite": Object {
                  "market_cap_usd": "2101330537.3210911914502786",
                  "max_supply": null,
                  "name": "Stellar",
                  "percent_change_24_hr": "-0.5797575094501702",
                  "price_usd": "0.1096095384438811",
                  "rank": "5",
                  "supply": "19171055431.4299000000000000",
                  "symbol": "XLM",
                },
                "removeFromFavorites": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
